plugins/rikki/loungestatistics/components/TeamStatistics.php

```diff
@@ -24,18 +24,46 @@ public function componentDetails()
     public $team = null;
     public $heroes = null;
     public $maps = null;
-    public $allseasons = null;
+    public $participatedSeasons = null;
+    public $selectedSeason = null;
 
     public function init()
     {
         $this->addJs('assets/js/datatables.min.js');
         $this->addJs('assets/js/loungestatistics.js');
         $this->addCss('assets/css/datatables.min.css');
+
         $this->team = Team::where('id', $this->property('team_id'))
                         ->with('matches', 'matches.games', 'matches.games.map', 'matches.games.gameParticipations', 'matches.games.gameParticipations.hero', 'matches.games.teamOneFirstBan', 'matches.games.teamOneSecondBan', 'matches.games.teamTwoFirstBan', 'matches.games.teamTwoSecondBan', 'matches.games.teamOneThirdBan', 'matches.games.teamTwoThirdBan')
                         ->first();
-        $this->allseasons = Season::where('reg_open', false)->get()->sortByDesc('created_at');
-        $this->calculateStats($this->allseasons->first());
+
+        $teamSeasonsSet = [];
+        foreach ($this->team->matches as $match) {
+            foreach ($match->associatedSeasons() as $season) {
+                if ($season) {
+                    $teamSeasonsSet[$season->id] = $season;
+
+                    if ($season->is_active && ($this->selectedSeason == null || $season->created_at < $this->selectedSeason->created_at)) {
+                        $this->selectedSeason = $season;
+                    }
+                }
+            }
+        }
+
+        $this->participatedSeasons = array_values($teamSeasonsSet);
+        usort($this->participatedSeasons, function($a, $b) {
+            if ($a->created_at == $b->created_at) return 0;
+            return $a->created_at > $b->created_at ? 1 : -1;
+        });
+
+        if ($this->selectedSeason == null) {
+            $participatedSeasonsCount = count($this->participatedSeasons);
+            if ($participatedSeasonsCount > 0) {
+                $this->selectedSeason = $this->participatedSeasons[$participatedSeasonsCount - 1];
+            }
+        }
```

Comment:
You could probably refactor most of this method and the one in slothstatistics into one - this duplicates a lot of code

Reply:
I did not consolidate them after my change to use collections because it would be more overhead to share code with their differences than to accept a bit of copy-paste